app/src/main/java/one/mixin/android/ui/conversation/FriendsFragment.kt

```diff
@@ -0,0 +1,97 @@
+import javax.inject.Inject
+
+class FriendsFragment : BaseFragment(), FriendAdapter.FriendListener {
+
+    companion object {
+        const val TAG = "FriendsFragment"
+        const val ARGS_IS_GROUP = "args_is_group"
+        const val ARGS_IS_BOT = "args_is_bot"
+
+        fun newInstance(conversationId: String, isGroup: Boolean, isBot: Boolean) = FriendsFragment().apply {
+            arguments = Bundle().apply {
+                putString(CONVERSATION_ID, conversationId)
+                putBoolean(ARGS_IS_GROUP, isGroup)
+                putBoolean(ARGS_IS_BOT, isBot)
+            }
+        }
+    }
+
+    @Inject
+    lateinit var viewModelFactory: ViewModelProvider.Factory
+    private val conversationViewModel: ConversationViewModel by lazy {
+        ViewModelProviders.of(this, viewModelFactory).get(ConversationViewModel::class.java)
+    }
+    @Inject
+    lateinit var jobManager: MixinJobManager
+
+    private val adapter = FriendAdapter().apply { listener = this@FriendsFragment }
+    private val conversationId: String by lazy { arguments!!.getString(CONVERSATION_ID) }
+    private val isGroup: Boolean by lazy { arguments!!.getBoolean(ARGS_IS_GROUP) }
+    private val isBot: Boolean by lazy { arguments!!.getBoolean(ARGS_IS_BOT) }
+
+    private var users = arrayListOf<User>()
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View =
+        layoutInflater.inflate(R.layout.fragment_friends, container, false)
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        title_view.left_ib.setOnClickListener {
+            search_et.hideKeyboard()
+            activity?.onBackPressed()
+        }
+        friends_rv.adapter = adapter
+        friends_rv.addItemDecoration(SpaceItemDecoration())
+        conversationViewModel.getFriends().observe(this, Observer {
+            if (it == null || it.isEmpty()) return@Observer
+
+            users.clear()
+            users.addAll(it)
+            adapter.friends = it
+        })
+
+        search_et.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {}
+
+            override fun afterTextChanged(s: Editable) {
+                val us = arrayListOf<User>()
+                users.forEach {
+                    if (it.fullName?.contains(s, true) == true) {
+                        us.add(it)
+                    }
+                }
+                adapter.friends = us
+            }
+        })
+    }
+
+    override fun onFriendClick(user: User) {
+        val fw = ForwardMessage(ForwardCategory.CONTACT.name, sharedUserId = user.userId)
+        ConversationActivity.show(context!!, conversationId, null, messages = arrayListOf(fw), isGroup = isGroup, isBot = isBot)
```

Comment:
Dont use !! unpack the optional context, use requireContext() get the NonNull context

Reply:
fixed